pkgs/by-name/sp/space-cadet-pinball/package.nix

```diff
@@ -6,71 +6,77 @@
   cmake,
   SDL2,
   SDL2_mixer,
-  Cocoa,
-  unrar-wrapper,
-  makeWrapper,
+  unrar,
 }:
 
 let
-  assets =
-    (fetchzip {
-      url = "https://archive.org/download/SpaceCadet_Plus95/Space_Cadet.rar";
-      sha256 = "sha256-fC+zsR8BY6vXpUkVd6i1jF0IZZxVKVvNi6VWCKT+pA4=";
-      stripRoot = false;
-    }).overrideAttrs
-      (old: {
-        nativeBuildInputs = old.nativeBuildInputs ++ [ unrar-wrapper ];
-      });
+  assets = fetchzip {
+    url = "https://archive.org/download/SpaceCadet_Plus95/Space_Cadet.rar";
+    hash = "sha256-fC+zsR8BY6vXpUkVd6i1jF0IZZxVKVvNi6VWCKT+pA4=";
+    stripRoot = false;
+    nativeBuildInputs = [ unrar ];
+  };
+  darwinApp = "$out/Applications/SpaceCadetPinball.app/Contents";
+  assetsDest =
+    if stdenv.hostPlatform.isDarwin then darwinApp + "/Resources" else "$out/share/SpaceCadetPinball";
 in
 stdenv.mkDerivation rec {
   pname = "SpaceCadetPinball";
   version = "2.1.0";
 
   src = fetchFromGitHub {
     owner = "k4zmu2a";
-    repo = pname;
+    repo = "SpaceCadetPinball";
     rev = "Release_${version}";
     hash = "sha256-W2P7Txv3RtmKhQ5c0+b4ghf+OMsN+ydUZt+6tB+LClM=";
   };
 
-  nativeBuildInputs = [
-    cmake
-    makeWrapper
-  ];
+  nativeBuildInputs = [ cmake ];
   buildInputs = [
     SDL2
     SDL2_mixer
-  ] ++ lib.optional stdenv.hostPlatform.isDarwin Cocoa;
+  ];
 
-  # Darwin needs a custom installphase since it is excluded from the cmake install
+  postPatch = ''
+    # Change the hardcoded FHS assets path
+    substituteInPlace SpaceCadetPinball/pch.h \
+      --replace-fail /usr/share ${placeholder "out"}/share
+    # Disable building a universal binary on Darwin, otherwise the cc wrapper passing -arch breaks the build
+    substituteInPlace CMakeLists.txt \
+      --replace-fail "arm64;x86_64" ""
+  '';
+
+  # Darwin needs a custom installPhase since it is excluded from the cmake install
   # https://github.com/k4zmu2a/SpaceCadetPinball/blob/0f88e43ba261bc21fa5c3ef9d44969a2a079d0de/CMakeLists.txt#L221
+  # This builds a bundle similar to what upstream's build script produces
```

Comment:
Could you link the upstream build script that I assume you have taken the steps out of?

That's useful in case it changes in the future.